### PR TITLE
Fix NPE issue and default values

### PR DESCRIPTION
**Pull request by Flanker32**
Fix npe when configuration is not complete. #550 #551
Fix init deployment issue #548
Fix default values #553 #549

### CODE REVIEW

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java [hunk 1]
```diff
@@ -184,17 +175,20 @@ private WebAppConfiguration getWebAppConfiguration(WebAppConfiguration configura
             NOT_EMPTY_REGEX, null, null);
 
         final String defaultRegion = configuration.region != null ? configuration.region.name() :
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java]: add EUROPE_WEST as a constant default value in configuration, provide a getRegion method returning with the failback-to-default code.

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java [hunk 2]
```diff
@@ -40,10 +40,10 @@
     public static final String NOT_EMPTY_REGEX = "[\\s\\S]+";
     public static final String BOOLEAN_REGEX = "[YyNn]";
 
-    public static final String CONFIGURATION_NOT_FOUND = "Configuration not found, init new configuration.";
-    public static final String CONVERT_TO_V2SCHEMA = "Converting configuration to V2 schema";
     public static final String SAVING_TO_POM = "Saving configuration to pom.";
-    public static final String EXITING = "Config only support V2 schema, exiting";
+    public static final String CONFIG_ONLY_SUPPORT_V2 = "Config only support V2 schema";
+    public static final String CHANGE_OS_WARNING = "The plugin may not work if you change the os of an existing " +
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java]: Why we provide this warning here?

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java [hunk 3]
```diff
@@ -104,30 +92,33 @@ protected boolean confirmConfiguration(WebAppConfiguration configuration) throws
...
-            case Docker:
-                System.out.println("Image : " + configuration.getImage());
-                System.out.println("ServerId : " + configuration.getServerId());
-                System.out.println("RegistryUrl : " + configuration.getRegistryUrl());
-                break;
-            default:
-                throw new MojoExecutionException("The value of <os> is unknown.");
+
+        if (configuration.getOs() == null) {
+            System.out.println("No runtime configuration");
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java]: What information do we hope the users to get here?

If the output is only `No runtime configuration`. Then this is much more like a log in debug mode.

Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java]: For we support user didn't config runtime and deploy, and for now we don't know whether user deploy to an existing webapp or just create a new one, so add this information to warn user they didn't complete the runtime if they plan to deploy a new webapp.

**Pull request merged**